mttl/models/containers/expert_containers.py

```diff
@@ -594,21 +593,10 @@ def add_expert(
             self.__layer_name__, expert.expert_weights
         )
 
-        if expert.name in self.experts:
-            raise ValueError(
-                "An expert with name {} already exists.".format(expert.name)
-            )
-
         if action == "merge":
             raise ValueError("Merging is not supported for `KVAdapters`.")
 
-        expert_config = ModifierConfig.from_training_config(expert.expert_config)
-        self._check_config(expert_config)
+        self._check_config(expert.expert_config)
 
         expert_module = KVAdapter(expert_config, self.attn_layer)
```

Comment:
Should `expert_config` be `expert.expert_config`?

`expert_module = KVAdapter(expert.expert_config, self.attn_layer)`

Reply:
done